### ISSUE TITLE
Load torchscript model with Libtorch c++ on Windows

### ISSUE THREAD

**Issue by jostrkr** [rusty1s/pytorch_geometric#3089]
Hi,

First, thanks for your amazing library!
I sucessfully froze the attached model thanks to torchscript and your jit support (.jittable() function)
[gcn.zip](https://github.com/rusty1s/pytorch_geometric/files/7116132/gcn.zip)
So I compiled torch-sparse and torch-scatter API with VS2017 and then I created this testing snippet:

main.cpp

    #include <iostream>
    #include <torch/torch.h>
    #include <torch/script.h>
    #include <torchsparse/sparse.h>
    #include <torchscatter/scatter.h>
    
     int main(int argc, const char* argv[]) {
    
    // Test torchsparse library
    torch::Tensor tensor = torch::tensor({ 0, 0, 0, 0, 1, 1, 1 });
    std::cout << tensor << std::endl;
    std::cout << ind2ptr(tensor, 2) << std::endl;
    
    // Test torchscatter library
    torch::Tensor src = torch::tensor({ 0.5, 0.4, 0.1, 0.6 });
    torch::Tensor index = torch::tensor({ 0, 0, 1, 1 });
    std::cout << src << std::endl;
    std::cout << index << std::endl;
    std::cout << scatter_sum(src, index, 0, torch::nullopt, torch::nullopt) << std::endl;
    
    // Test frozen GNN loading
    if (argc != 2) {
	    std::cerr << "usage: example-app <path-to-exported-script-module>\n";
	    return -1;
    }
    
    std::cout << "Trying to load model..." << std::endl;
    torch::jit::script::Module module;
    
    try {
	    module = torch::jit::load(argv[1]);
    }
    catch (const c10::Error& e) {
	    std::cout << e.what() << std::endl;
	    std::cerr << "error loading the model\n";
	    return -1;
    }
    
    // Test inference forward pass
    auto x = torch::tensor({ {-1}, {0}, {1} }, at::kFloat);
    auto edge_index = torch::tensor({ {0, 1, 1, 2}, {1, 0, 2, 1} }, at::kLong);
    auto edge_weight = torch::tensor({ 0, 1, 1, 2 }, at::kFloat);
    auto batch = torch::tensor({ 0, 0, 0 }, at::kLong);
    
    auto output = module({ x, edge_index, edge_weight, batch }).toTensor();
    std::cout << "output: " << output << std::endl;
    }

CMakeLists.txt

    cmake_minimum_required(VERSION 3.10)
    project(hello-world)
    
    find_package(TorchSparse REQUIRED)
    find_package(TorchScatter REQUIRED)
    
    add_executable(hello-world main.cpp)
    
    target_compile_features(hello-world PUBLIC cxx_range_for)
    target_link_libraries(hello-world TorchSparse::TorchSparse TorchScatter::TorchScatter)
    set_property(TARGET hello-world PROPERTY CXX_STANDARD 14)

Unfortunately, I cannot load the model in C++ side with LibTorch (I use cpu version 1.8.1). I copied the DLLs torchsparse.dll/torchscatter.dll next to my application exe but it crashes with a torch::jit::script::ErrorReport.
However, I also tested the deployment on Ubuntu 18.04 and actually it worked with libtorchsparse.so and libtorchscatter.so (built with gcc) i.e. the model is successfully loaded on C++ side and a single forward pass with dummy inputs also worked! So I guess it might be an issue linked to Windows specifically... By the way, I opened a thread on pytorch repo (see https://github.com/pytorch/pytorch/issues/64490).
Did you have any success deploying to production a GNN torchscript model on Windows with custom operators? Thanks in advance!

**Comment by rusty1s** [comment]
I have sadly no experience with loading TorchScript models on Windows. Do the tests calls at the beginning of your script succeed? Do you have more error information besides `torch::jit::script::ErrorReport`?

**Comment by jostrkr** [comment]
Yes, the tests passed. I had to compile pytorch_scatter and pytorch_sparse as a static library to get the symbols in the respective .lib files though. Otherwise, I missed some symbols (the cpp implementation was not contained in the .lib files). I simply changed in the CMakeLists.txt:
add_library(${PROJECT_NAME} SHARED ${OPERATOR_SOURCES}) -> add_library(${PROJECT_NAME} STATIC ${OPERATOR_SOURCES}).
This way I was able to get the .lib AND .dll for each library. Just for info, on the other side, only .so files are needed on Ubuntu to make it work.

I do not have much information about the torch::jit::script::ErrorReport since I compiled in ReleaseWithDebInfo mode, just a memory violation but I might compile in debug mode to give it a try.

**Comment by jostrkr** [comment]
Just to let you know, I compiled everything in debug mode but sadly the torch::jit::script::ErrorReport error does not show any relevant information except a memory access violation...

**Comment by rusty1s** [comment]
Thanks. Let's find out if the PyTorch team has some more pointers on how to proceed with this.

**Comment by akihironitta** [comment]
Closing this issue because TorchScript has been unmaintained by the PyTorch team for 3+ years, and PyTorch 2.9 has officially deprecated its support in favor of PT2 stack.
